From a40287727d9b737e183959fd31a4e0c55f312853 Mon Sep 17 00:00:00 2001
From: Erel Geron <erelx.geron@intel.com>
Date: Mon, 28 May 2018 17:15:56 +0300
Subject: [PATCH] iwlwifi: fix non_shared_ant for 22000 devices
Git-commit: a40287727d9b737e183959fd31a4e0c55f312853
Patch-mainline: v4.20
References: bsc#1119086

The non-shared antenna was wrong for 22000 device series.
Fix it to ANT_B for correct antenna preference by coex in MVM driver.

Fixes: e34d975e40ff ("iwlwifi: Add a000 HW family support")
Signed-off-by: Erel Geron <erelx.geron@intel.com>
Signed-off-by: Luca Coelho <luciano.coelho@intel.com>
Signed-off-by: Oliver Neukum <oneukum@suse.com>
---
 drivers/net/wireless/intel/iwlwifi/cfg/a000.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/wireless/intel/iwlwifi/cfg/a000.c
+++ b/drivers/net/wireless/intel/iwlwifi/cfg/a000.c
@@ -118,7 +118,7 @@ static const struct iwl_ht_params iwl_a0
 	.base_params = &iwl_a000_base_params,				\
 	.led_mode = IWL_LED_RF_STATE,					\
 	.nvm_hw_section_num = NVM_HW_SECTION_NUM_FAMILY_A000,		\
-	.non_shared_ant = ANT_A,					\
+	.non_shared_ant = ANT_B,					\
 	.dccm_offset = IWL_A000_DCCM_OFFSET,				\
 	.dccm_len = IWL_A000_DCCM_LEN,					\
 	.dccm2_offset = IWL_A000_DCCM2_OFFSET,				\
